app/controllers/foreman_bootdisk/disks_controller.rb

```diff
@@ -60,5 +65,19 @@ def help; end
     def resource_scope(_controller = controller_name)
       Host::Managed.authorized(:view_hosts)
     end
+
+    def error_rendering(exception)
+      msg = _('Failed to render boot disk template')
+      error("#{msg}: #{exception.message}")
+      ::Foreman::Logging.exception(msg, exception)
+    end
+
+    def prolong_token(host)
+      return if Setting[:token_duration] == 0 || host.token.nil?
+
+      # update build token
+      token = host.token
+      token.update(expires: token.expires + Setting[:token_duration].minutes)
```

Comment:
There is a bug, on every download it adds token duration to the expiration, so in keeps adding on every request. You need to set expiration in relative to the current time and not the last seen expiration.